src/ILCompiler.MetadataWriter/src/Internal/Metadata/NativeFormat/Writer/LowLevelLinq.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
```

Comment:
There is actually a similar light-weight linq implementation at `src\Common\src\System\Linq\LowLevelEnumerable.cs` that it would be nice to unify this with. Partial class split into a few .cs files, e.g. AsList method should be separate to avoid introducing bad dependencies.

The naming convention for `LowLevel*` has been to use the same namespace as the full blown counterpart, but I guess that it does not work well here because of you get a reference to System.Linq.dll via standard library. We would need to find a way to remove the reference to System.Linq.dll to avoid ambiguity - likely go back to explicit list of references.

Or if unique namespace is better - I would call it `System.LowLevelLinq` to make it follow the naming convention.